python/lsst/pipe/base/graphBuilder.py

```diff
@@ -1123,18 +1149,19 @@ def resolveDatasetRefs(
                 # the quantum because those should never be resolved.
                 if skip_collections_wildcard is not None or (run_exists and clobberOutputs):
                     resolvedRefs = []
-                    unresolvedRefs = []
+                    unresolvedDataIds = []
                     haveMetadata = False
                     for datasetType, originalRefs in quantum.outputs.items():
-                        for ref in task.outputs.extract(datasetType, originalRefs.keys()):
-                            if ref.id is not None:
+                        for dataId, ref in task.outputs.extract(datasetType, originalRefs.keys()):
+                            if ref is not None:
                                 resolvedRefs.append(ref)
+                                originalRefs[dataId].ref = ref
```

Comment:
@TallJimbo, this is the line that fixes disconnected graphs (would read `originalRefs[dataId] = ref` in old code). With _RefHolder thing in this new implementation only the pure outputs need this line, intermediates are updated because _refHolders are actually shared between quanta. In the old code refs were not shared, so this line would also update intermediates.

Reply:
Ah, thanks.

And now that you've pointed me at this, I see a comment a file lines up from this point in the diff that mentions refs that are "never resolved"; that should get updated here.

Reply:
Hmm, I never read the comments 🙁 I probably need to update that comment too, it's confusing now.

Reply:
I removed that sentence.